### PR TITLE
Allow parent agency users and SRES users to set project agency

### PR DESCRIPTION
**Pull request by goemen**
On submit and edit project the ability to change the projects agency so that I can submit properties on their behalf 

User has to be at the Ministry Level to be able to do this. They can only select their agency or sub agency

OR 

If they are SRES they can select any agency or sub agency 

PROPERTIES in the list must belong to the selected agency

### CODE REVIEW

File: frontend/src/features/projects/common/forms/ProjectDraftForm.tsx [hunk 1]
```diff
@@ -55,6 +107,19 @@ const ProjectDraftForm = ({
...
       </Form.Row>
+      {(isSRES || isUserAgencyAParent) && (
+        <Form.Row className="col-md-10">
+          <Form.Label className="col-md-1">Project Agency</Form.Label>
+          <AgencyCol className="col-md-5">
+            <ParentSelect
+              field={'agencyId'}
+              options={agencyOptions}
+              filterBy={['code', 'label', 'parent']}
+              convertValue={Number}
```
Comment (jdtoombs) [inline frontend/src/features/projects/common/forms/ProjectDraftForm.tsx:118]: Ooh good idea, much easier on the eyes

**Review by jdtoombs — APPROVED** [review]

File: backend/tests/unit/dal/Services/ProjectServiceTest.cs [hunk 2]
```diff
@@ -442,6 +442,38 @@ public async void Add_Project()
             queueService.Verify(m => m.NotificationQueue.SendNotificationsAsync(It.IsAny<IEnumerable<NotificationQueue>>(), It.IsAny<bool>()), Times.Once());
         }
 
+        /// <summary>
+        /// User cannot add project to an agency that they do not belong to.
+        /// </summary>
+        [Fact]
+        public async void AddProject_NoInAgency()
```
Comment (devinleighsmith) [inline backend/tests/unit/dal/Services/ProjectServiceTest.cs:449]: /nit NotInAgency

File: frontend/src/components/common/form/ParentSelect.tsx [hunk 3]
```diff
@@ -50,6 +52,7 @@ export const ParentSelect: React.FC<IParentSelect> = ({
   label,
   onChange,
   selectClosest,
+  convertValue = value => value,
```
Comment (devinleighsmith) [inline frontend/src/components/common/form/ParentSelect.tsx:55]: Is this doing anything?

Comment (devinleighsmith) [reply frontend/src/components/common/form/ParentSelect.tsx:55]: I think I'm missing something here.

File: frontend/src/features/projects/common/forms/ProjectDraftForm.tsx [hunk 4]
```diff
@@ -24,6 +49,33 @@ const ProjectDraftForm = ({
   title,
   setIsReadOnly,
 }: IStepProps & IProjectDraftFormProps) => {
+  const { getOptionsByType } = useCodeLookups();
+  const keycloak = useKeycloakWrapper();
+  const agencies = getOptionsByType(API.AGENCY_CODE_SET_NAME);
+  const userAgency = agencies.find(a => Number(a.value) === Number(keycloak.agencyId));
+  const { values } = useFormikContext();
+  console.log(values);
```
Comment (devinleighsmith) [inline frontend/src/features/projects/common/forms/ProjectDraftForm.tsx:57]: console.log

File: frontend/src/components/common/form/ParentSelect.tsx (hunk 3 above)
Comment (devinleighsmith) [reply frontend/src/components/common/form/ParentSelect.tsx:55]: maybe we need a generic here to type the underlying object?

**Review by devinleighsmith — APPROVED** [review]

File: frontend/src/components/common/form/ParentSelect.tsx (hunk 3 above)
Comment (goemen) [reply frontend/src/components/common/form/ParentSelect.tsx:55]: I added an optional callback to transform the value before update formik value, this line is just defaulting to the raw value form the input.

**Pull request merged**